### PR TITLE
Remove support for Ruby versions <= 2.1

### PR DESCRIPTION
**Pull request by xtrasimplicity**
Given Cucumber-ruby 3.0.2 removes support for Ruby versions <= 2.1 and these particular versions are making the test suite very brittle, I propose we remove support starting from the next release (1.6).

Does anyone have any thoughts on this?

Thanks!

### CODE REVIEW

**Comment by olleolleolle** [comment]
I like this change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,8 @@ Please see [CONTRIBUTING.md](https://github.com/cucumber/cucumber/blob/master/CO
 ### Changed
  * Renamed History.md to CHANGELOG.md, added contributing note, and this line in accordance with [cucumber/cucumber #251](https://github.com/cucumber/cucumber/issues/251) ([#345](https://github.com/cucumber/cucumber-rails/pull/345) [jaysonesmith](https://github.com/jaysonesmith))
  * Update .travis.yml with ruby versions ([#341](https://github.com/cucumber/cucumber-rails/pull/341) Jun Aruga)
-
+ * Removed support for Ruby <= 2.1, to keep in line with the [core Cucumber-ruby repository](https://github.com/cucumber/cucumber-ruby/blob/master/CHANGELOG.md#302-2017-11-11).
```
Comment (olleolleolle) [inline CHANGELOG.md]: Tiny detail: `cucumber-ruby` is the name of the repository. We can refer to that casing of the name in the text.

File: cucumber-rails.gemspec [hunk 2]
```diff
@@ -36,6 +36,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency('rdoc', '>= 3.4')
   s.add_development_dependency('yard', '>= 0.8.7')
 
+  s.required_ruby_version = '>= 2.2.0'
```
Comment (olleolleolle) [inline cucumber-rails.gemspec]: This is super. So helpful.

File: CHANGELOG.md (hunk 1 above)
Comment (xtrasimplicity) [reply CHANGELOG.md]: Thanks @olleolleolle, I'll make that change now and merge the request. I've also noticed that I've set the minimum cucumber version to 3.0.0 where it should really be 3.0.2 (given that is when Cucumber-ruby officially ceases support for Ruby <= 2.1) - I'll fix this, too.

**Pull request merged**